src/main/java/org/apache/mesos/scheduler/recovery/SimpleRecoveryPlanManager.java

```diff
@@ -0,0 +1,325 @@
+        LOGGER.info("Ignoring setPlan call. {} generates plans dynamically.", getClass().getName());
+    }
+
+    /**
+     * Returns the first {@link Phase} in the {@link Plan} which isn't marked complete.
+     */
+    @Override
+    public Optional<Phase> getCurrentPhase() {
+        for (final Phase phase : plan.getPhases()) {
+            if (!phase.isComplete()) {
+                LOGGER.debug("Phase {} ({}) is NOT complete. This is the current phase.",
+                        phase.getName(), phase.getId());
+                return Optional.of(phase);
+            } else {
+                LOGGER.debug("Phase {} ({}) is complete.", phase.getName(), phase.getId());
+            }
+        }
+        LOGGER.debug("All recovery phases are complete.");
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<Block> getCurrentBlock(List<Block> dirtiedAssets) {
+        refreshPlan(dirtiedAssets);
+
+        if (phaseStrategy != null) {
+            final Optional<Block> chosenBlock = phaseStrategy.getCurrentBlock();
+            if (chosenBlock.isPresent() &&
+                    launchConstrainer.canLaunch(((DefaultRecoveryBlock) chosenBlock.get()).getRecoveryRequirement())) {
+                return Optional.of(chosenBlock.get());
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public boolean isComplete() {
+        return plan != null && plan.isComplete();
+    }
+
+    @Override
+    public void proceed() {
+        LOGGER.info("Resuming recovery...");
+        if (phaseStrategy != null) {
+            phaseStrategy.proceed();
+            LOGGER.info("Done resuming recovery for phase: {}", phaseStrategy.getPhase());
+        } else {
+            LOGGER.info("No phase to resume");
+        }
+    }
+
+    @Override
+    public void interrupt() {
+        LOGGER.info("Interrupting recovery");
+        if (phaseStrategy != null) {
+            phaseStrategy.interrupt();
+            LOGGER.info("Interrupted current phase: phase = {}", phaseStrategy.getPhase());
+        } else {
+            LOGGER.info("No phase to interrupt");
+        }
+    }
+
+    @Override
+    public boolean isInterrupted() {
+        return getStatus() == Status.WAITING;
+    }
+
+    @Override
+    public void restart(UUID phaseId, UUID blockId) {
+        if (phaseStrategy != null) {
+            phaseStrategy.restart(blockId);
+        }
+    }
+
+    @Override
+    public void forceComplete(UUID phaseId, UUID blockId) {
+        if (phaseStrategy != null) {
+            phaseStrategy.forceComplete(blockId);
+        }
+    }
+
+    @Override
+    public void update(Protos.TaskStatus status) {
+        LOGGER.info("Ignoring status: {}", TextFormat.shortDebugString(status));
+    }
+
+    @Override
+    public boolean hasDecisionPoint(Block block) {
+        return phaseStrategy != null && phaseStrategy.hasDecisionPoint(block);
+    }
+
+    @Override
+    public Status getStatus() {
```

Comment:
This is slightly special:
`} else if (plan == null || CollectionUtils.isEmpty(plan.getPhases())) {` (SimpleRecoveryPlanManager)

vs

`} else if (plan.getPhases().isEmpty()) {` (DefaultPlanManager)

But, I think `SimpleRecoveryPlanManager` version would also work for `DefaultPlanManager`.

Let me move this to `PlanManager` interface as the default impl for `getStatus()` method.